skfda/representation/irregular.py

```diff
@@ -1223,61 +1227,45 @@ def copy(  # noqa: WPS211
     def restrict(  # noqa: WPS210
         self: T,
         domain_range: DomainRangeLike,
+        *,
+        with_bounds: bool = False,
     ) -> T:
         """
         Restrict the functions to a new domain range.
 
         Args:
             domain_range: New domain range.
+            with_bounds: Whether or not to ensure domain boundaries
+                appear in `grid_points`.
 
         Returns:
             T: Restricted function.
 
         """
+        if with_bounds:  # To do
+            raise NotImplementedError('Not yet implemented for FDataIrregular')
+
         from ..misc.validation import validate_domain_range
 
-        domain_range = validate_domain_range(domain_range)
+        npdr = np.asarray(validate_domain_range(domain_range))  # shape(dim, 2)
 
-        head = 0
-        indices = []
-        arguments = []
-        values = []
-        sample_names = []
-
-        # Eliminate points outside the new range.
-        # Must also modify function indices to point to new array
-
-        slice_points = np.split(self.points, self.start_indices[1:])
-        slice_values = np.split(self.values, self.start_indices[1:])
-
-        for i, points_values in enumerate(zip(slice_points, slice_values)):
-            sample_points, sample_values = points_values
-            masks = set(range(sample_points.shape[0]))
-            for dim, dr in enumerate(domain_range):
-                dr_start, dr_end = dr
-                select_mask = np.where(
-                    (
-                        (dr_start <= sample_points[:, dim])
-                        & (sample_points[:, dim] <= dr_end)
-                    ),
-                )
+        mask = np.all(
+            (npdr[:, 0] <= sample_points) & (sample_points <= npdr[:, 1]),
```

Comment:
`sample_points` is not defined and this is why the tests are failing.

Reply:
Thanks, fixed!
Also, if `dim_domain > 1` and user passes only `domain=(a, b)`, now it broadcasts to `((a, b), (a, b), ...)`.